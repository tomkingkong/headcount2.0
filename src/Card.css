.Card {
  cursor: pointer;
  background-color: #fff;
  border-radius: 0.2rem;
  box-shadow: 0.1rem 0.1rem lightgrey;
  overflow: scroll;
  min-width: 8rem;
  width: 100%;
  max-width: 10rem;
  margin: 0.5rem 0.5rem;
  padding: 0.5rem;
  transition: all 0.25s linear;
}

.Card:hover {
  box-shadow: 0.4rem 0.4rem lightblue;
}

.Card ul {
  list-style-type: none;
  padding: auto;
}

.Card h1 {
  border-bottom: 0.2rem double transparent;
  font-size: 0.9rem;
  padding-bottom: 0.5rem;
  height: 1.8rem;
  text-align: center;
  transition: all 0.35s linear;
}

.Card:hover h1 {
  border-bottom: 0.2rem double lightblue;
  color: rgba(8, 107, 221);
}

.selected {
  border: 0.2rem solid rgb(18, 100, 194);
}

.Card .Green {
  color: rgb(12, 173, 12);
}

.Card .Red {
  color: rgb(243, 14, 14);
}