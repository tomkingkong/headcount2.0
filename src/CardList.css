.CardList {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  min-width: 10rem;
  margin: 0 auto;
}