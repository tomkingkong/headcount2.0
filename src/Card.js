import React from 'react';

import './Card.css';

const Card = ({stats, location}) => {
  const dates = Object.keys(stats).map(year => <li className={stats[year]>0.5 ? 'Green' : 'Red'} key={location+year}>{year}: {stats[year]}</li>)
  
  return (
    <article className="Card">
      <h1>{location}</h1>
      <ul>{dates}</ul>
    </article>
  );
}

export default Card;