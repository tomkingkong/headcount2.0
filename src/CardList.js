import React from 'react';
import PropTypes from 'prop-types';

import './CardList.css';
import Card from './Card';

const CardList = ({dataCards, toggleCompare}) => {
  const displayCards = dataCards.map(card => (
    <Card {...card} 
      key={card.location}
      id={card.location} 
      toggleCompare={toggleCompare} 
    />
  ));

  return (
    <section className="CardList">
      {displayCards}
    </section>
  );
};

CardList.propTypes = {
  dataCards: PropTypes.array,
  toggleCompare: PropTypes.func
};

export default CardList;