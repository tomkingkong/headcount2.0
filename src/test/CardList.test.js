import React from 'react';
import { shallow, mount } from 'enzyme';

import Card from '../Card';
import CardList from '../CardList.js';

const mockData = [
  {
    'stats': {
      2005: 0.521, 
      2017: 0.521, 
      2018: 0.521, 
      2019: 0.521 
    }, 
    'location': 'COLORADO', 
    'id':'COLORADO'
  },
  {
    'stats': {
      2005: 0.521, 
      2017: 0.521, 
      2018: 0.521, 
      2019: 0.521 
    }, 
    'location': 'ASPEN', 
    'id':'ASPEN'
  },
  {
    'stats': {
      2005: 0.521, 
      2017: 0.521, 
      2018: 0.521, 
      2019: 0.521 
    }, 
    'location': 'SHMOOP', 
    'id':'SHMOOP'
  }
];

describe('CardList', () => {

  beforeEach(() => {
    wrapper = shallow(<CardList cards={mockData} />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('renders without crashing', () => {
    const section = document.createElement('section');
    ReactDOM.render(<CardList cards={mockData} />, section);
  });

  it('should render section element', () => {
    expect(wrapper.find('section').length).toEqual(1);
  });

  it('should display correct number of cards', () => {
    wrapper = mount(<CardList cards={mockData} />)
    expect(wrapper.find('article').length).toEqual(3);
  });
});