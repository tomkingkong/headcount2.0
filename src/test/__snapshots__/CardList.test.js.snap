// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList should match the snapshot passing an array with data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    dataCards={
        Array [
            Object {
              "id": "COLORADO",
              "location": "COLORADO",
              "stats": Object {
                "2005": 0.521,
                "2017": 0.521,
                "2018": 0.521,
                "2019": 0.521,
              },
            },
            Object {
              "id": "ASPEN",
              "location": "ASPEN",
              "stats": Object {
                "2005": 0.521,
                "2017": 0.521,
                "2018": 0.521,
                "2019": 0.521,
              },
            },
            Object {
              "id": "SHMOOP",
              "location": "SHMOOP",
              "stats": Object {
                "2005": 0.521,
                "2017": 0.521,
                "2018": 0.521,
                "2019": 0.521,
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          compare={undefined}
          compareIDs={undefined}
          id="COLORADO"
          location="COLORADO"
          stats={
                    Object {
                              "2005": 0.521,
                              "2017": 0.521,
                              "2018": 0.521,
                              "2019": 0.521,
                            }
          }
          toggleCompare={undefined}
/>,
        <Card
          compare={undefined}
          compareIDs={undefined}
          id="ASPEN"
          location="ASPEN"
          stats={
                    Object {
                              "2005": 0.521,
                              "2017": 0.521,
                              "2018": 0.521,
                              "2019": 0.521,
                            }
          }
          toggleCompare={undefined}
/>,
        <Card
          compare={undefined}
          compareIDs={undefined}
          id="SHMOOP"
          location="SHMOOP"
          stats={
                    Object {
                              "2005": 0.521,
                              "2017": 0.521,
                              "2018": 0.521,
                              "2019": 0.521,
                            }
          }
          toggleCompare={undefined}
/>,
      ],
      "className": "CardList",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "COLORADO",
        "nodeType": "class",
        "props": Object {
          "compare": undefined,
          "compareIDs": undefined,
          "id": "COLORADO",
          "location": "COLORADO",
          "stats": Object {
            "2005": 0.521,
            "2017": 0.521,
            "2018": 0.521,
            "2019": 0.521,
          },
          "toggleCompare": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "ASPEN",
        "nodeType": "class",
        "props": Object {
          "compare": undefined,
          "compareIDs": undefined,
          "id": "ASPEN",
          "location": "ASPEN",
          "stats": Object {
            "2005": 0.521,
            "2017": 0.521,
            "2018": 0.521,
            "2019": 0.521,
          },
          "toggleCompare": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "SHMOOP",
        "nodeType": "class",
        "props": Object {
          "compare": undefined,
          "compareIDs": undefined,
          "id": "SHMOOP",
          "location": "SHMOOP",
          "stats": Object {
            "2005": 0.521,
            "2017": 0.521,
            "2018": 0.521,
            "2019": 0.521,
          },
          "toggleCompare": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            compare={undefined}
            compareIDs={undefined}
            id="COLORADO"
            location="COLORADO"
            stats={
                        Object {
                                    "2005": 0.521,
                                    "2017": 0.521,
                                    "2018": 0.521,
                                    "2019": 0.521,
                                  }
            }
            toggleCompare={undefined}
/>,
          <Card
            compare={undefined}
            compareIDs={undefined}
            id="ASPEN"
            location="ASPEN"
            stats={
                        Object {
                                    "2005": 0.521,
                                    "2017": 0.521,
                                    "2018": 0.521,
                                    "2019": 0.521,
                                  }
            }
            toggleCompare={undefined}
/>,
          <Card
            compare={undefined}
            compareIDs={undefined}
            id="SHMOOP"
            location="SHMOOP"
            stats={
                        Object {
                                    "2005": 0.521,
                                    "2017": 0.521,
                                    "2018": 0.521,
                                    "2019": 0.521,
                                  }
            }
            toggleCompare={undefined}
/>,
        ],
        "className": "CardList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "COLORADO",
          "nodeType": "class",
          "props": Object {
            "compare": undefined,
            "compareIDs": undefined,
            "id": "COLORADO",
            "location": "COLORADO",
            "stats": Object {
              "2005": 0.521,
              "2017": 0.521,
              "2018": 0.521,
              "2019": 0.521,
            },
            "toggleCompare": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "ASPEN",
          "nodeType": "class",
          "props": Object {
            "compare": undefined,
            "compareIDs": undefined,
            "id": "ASPEN",
            "location": "ASPEN",
            "stats": Object {
              "2005": 0.521,
              "2017": 0.521,
              "2018": 0.521,
              "2019": 0.521,
            },
            "toggleCompare": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "SHMOOP",
          "nodeType": "class",
          "props": Object {
            "compare": undefined,
            "compareIDs": undefined,
            "id": "SHMOOP",
            "location": "SHMOOP",
            "stats": Object {
              "2005": 0.521,
              "2017": 0.521,
              "2018": 0.521,
              "2019": 0.521,
            },
            "toggleCompare": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`CardList should match the snapshot passing an array with data objects and a function 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    dataCards={
        Array [
            Object {
              "id": "COLORADO",
              "location": "COLORADO",
              "stats": Object {
                "2005": 0.521,
                "2017": 0.521,
                "2018": 0.521,
                "2019": 0.521,
              },
            },
            Object {
              "id": "ASPEN",
              "location": "ASPEN",
              "stats": Object {
                "2005": 0.521,
                "2017": 0.521,
                "2018": 0.521,
                "2019": 0.521,
              },
            },
            Object {
              "id": "SHMOOP",
              "location": "SHMOOP",
              "stats": Object {
                "2005": 0.521,
                "2017": 0.521,
                "2018": 0.521,
                "2019": 0.521,
              },
            },
          ]
    }
    toggleCompare={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          compare={undefined}
          compareIDs={undefined}
          id="COLORADO"
          location="COLORADO"
          stats={
                    Object {
                              "2005": 0.521,
                              "2017": 0.521,
                              "2018": 0.521,
                              "2019": 0.521,
                            }
          }
          toggleCompare={[Function]}
/>,
        <Card
          compare={undefined}
          compareIDs={undefined}
          id="ASPEN"
          location="ASPEN"
          stats={
                    Object {
                              "2005": 0.521,
                              "2017": 0.521,
                              "2018": 0.521,
                              "2019": 0.521,
                            }
          }
          toggleCompare={[Function]}
/>,
        <Card
          compare={undefined}
          compareIDs={undefined}
          id="SHMOOP"
          location="SHMOOP"
          stats={
                    Object {
                              "2005": 0.521,
                              "2017": 0.521,
                              "2018": 0.521,
                              "2019": 0.521,
                            }
          }
          toggleCompare={[Function]}
/>,
      ],
      "className": "CardList",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "COLORADO",
        "nodeType": "class",
        "props": Object {
          "compare": undefined,
          "compareIDs": undefined,
          "id": "COLORADO",
          "location": "COLORADO",
          "stats": Object {
            "2005": 0.521,
            "2017": 0.521,
            "2018": 0.521,
            "2019": 0.521,
          },
          "toggleCompare": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "ASPEN",
        "nodeType": "class",
        "props": Object {
          "compare": undefined,
          "compareIDs": undefined,
          "id": "ASPEN",
          "location": "ASPEN",
          "stats": Object {
            "2005": 0.521,
            "2017": 0.521,
            "2018": 0.521,
            "2019": 0.521,
          },
          "toggleCompare": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "SHMOOP",
        "nodeType": "class",
        "props": Object {
          "compare": undefined,
          "compareIDs": undefined,
          "id": "SHMOOP",
          "location": "SHMOOP",
          "stats": Object {
            "2005": 0.521,
            "2017": 0.521,
            "2018": 0.521,
            "2019": 0.521,
          },
          "toggleCompare": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            compare={undefined}
            compareIDs={undefined}
            id="COLORADO"
            location="COLORADO"
            stats={
                        Object {
                                    "2005": 0.521,
                                    "2017": 0.521,
                                    "2018": 0.521,
                                    "2019": 0.521,
                                  }
            }
            toggleCompare={[Function]}
/>,
          <Card
            compare={undefined}
            compareIDs={undefined}
            id="ASPEN"
            location="ASPEN"
            stats={
                        Object {
                                    "2005": 0.521,
                                    "2017": 0.521,
                                    "2018": 0.521,
                                    "2019": 0.521,
                                  }
            }
            toggleCompare={[Function]}
/>,
          <Card
            compare={undefined}
            compareIDs={undefined}
            id="SHMOOP"
            location="SHMOOP"
            stats={
                        Object {
                                    "2005": 0.521,
                                    "2017": 0.521,
                                    "2018": 0.521,
                                    "2019": 0.521,
                                  }
            }
            toggleCompare={[Function]}
/>,
        ],
        "className": "CardList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "COLORADO",
          "nodeType": "class",
          "props": Object {
            "compare": undefined,
            "compareIDs": undefined,
            "id": "COLORADO",
            "location": "COLORADO",
            "stats": Object {
              "2005": 0.521,
              "2017": 0.521,
              "2018": 0.521,
              "2019": 0.521,
            },
            "toggleCompare": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "ASPEN",
          "nodeType": "class",
          "props": Object {
            "compare": undefined,
            "compareIDs": undefined,
            "id": "ASPEN",
            "location": "ASPEN",
            "stats": Object {
              "2005": 0.521,
              "2017": 0.521,
              "2018": 0.521,
              "2019": 0.521,
            },
            "toggleCompare": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "SHMOOP",
          "nodeType": "class",
          "props": Object {
            "compare": undefined,
            "compareIDs": undefined,
            "id": "SHMOOP",
            "location": "SHMOOP",
            "stats": Object {
              "2005": 0.521,
              "2017": 0.521,
              "2018": 0.521,
              "2019": 0.521,
            },
            "toggleCompare": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`CardList should match the snapshot passing an empty array 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    dataCards={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "CardList",
    },
    "ref": null,
    "rendered": Array [],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "CardList",
      },
      "ref": null,
      "rendered": Array [],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
